File: app/src/main/java/mydebts/android/app/data/ParticipantsSource.java
package mydebts.android.app.data;

import android.support.annotation.NonNull;

import java.util.List;

import io.reactivex.Single;
import mydebts.android.app.data.model.Participant;

public interface ParticipantsSource {

    Single<List<Participant>> getByEventId(@NonNull Long eventId);

    Single<List<Participant>> getByPersonId(@NonNull Long personId);

    Single<Participant> insert(@NonNull Participant participant);

    Single<Integer> deleteByEventId(@NonNull Long eventId);
}

File: app/src/main/java/mydebts/android/app/data/ParticipantsSource.kt
package mydebts.android.app.data

import io.reactivex.Single
import mydebts.android.app.data.model.Participant

interface ParticipantsSource {

    fun getByEventId(eventId: Long): Single<List<Participant>>

    fun getByPersonId(personId: Long): Single<List<Participant>>

    fun insert(participant: Participant): Single<Participant>

    fun update(participant: Participant): Single<Participant>

    fun delete(participant: Participant): Single<Participant>
}
